### PR TITLE
[BUGFIX] Added missing checkbox fields, added missing query parameters

### PR DESCRIPTION
**Pull request by barthy-koeln**
Hi again,

This PR contains a change to add `checkbox` fields as booleans to the schema.

I've also added the base REST query parameters, [as described in the payload documentation](https://payloadcms.com/docs/rest-api/overview).

May I ask what your plans are for this repository? 
I'm trying to gauge how much effort I should put into the current code base vs. writing or refactoring my own, if anything else comes up.

If you prefer discussing this separately from this PR, feel free to add me on Discord `barthy.koeln#2946`.
Sorry if I seem pushy, I'm just working on a Payload + Astro project right now :)

Barthy

### CODE REVIEW

File: src/openapiSpec.ts [hunk 1]
```diff
@@ -44,6 +43,12 @@ const globalName = (global: SanitizedGlobalConfig): string => {
 
 type ComponentType = 'schemas' | 'responses' | 'requestBodies'
 
+const baseQueryParams: Array<OpenAPIV3.ParameterObject & OpenAPIV3_1.ParameterObject> = [
+  { in: 'query', name: 'depth', schema: { type: 'number' } },
+  { in: 'query', name: 'locale', schema: { type: 'string' } },
+  { in: 'query', name: 'fallback-locale', schema: { type: 'string' } },
```
Comment (Teyras) [inline src/openapiSpec.ts:49]: I believe that available locales could (and eventually should) be extracted from the Payload config. Then, we could expose them as `{type: 'string', enum: ['en', ...]}`. If you just need a quick patch now, please make an issue to track this. Or just a code TODO.

File: src/openapiSpec.ts
Comment (Teyras) [inline src/openapiSpec.ts:1]: I believe the chmod change is not intentional?

**Review by Teyras — APPROVED** [review]
This is super awesome, thanks!

File: src/openapiSpec.ts (hunk 1 above)
Comment (barthy-koeln) [reply src/openapiSpec.ts:49]: Good idea. I have a local setup for my project right now, so there's no hurry to release on NPM.

File: src/openapiSpec.ts
Comment (barthy-koeln) [reply src/openapiSpec.ts:1]: Definitely not. I usually set git to ignore file modes. I will set it back to 644.

**Comment by Teyras** [comment]
> May I ask what your plans are for this repository? I'm trying to gauge how much effort I should put into the current code base vs. writing or refactoring my own, if anything else comes up.

Originally, I started it because there was no open source headless CMS with good OpenAPI support and PayloadCMS seemed like a cool project with a lot of momentum (well, it is exactly that).

I want to use the package for a couple of website projects, which will probably make me work on this package further. That will probably happen in two or three weeks, but no promises.

In general, my plans are outlined in the roadmap in the readme file. I understand it's very brief though. If you have some ideas or requests, feel free to open issues :slightly_smiling_face: 

> If you prefer discussing this separately from this PR, feel free to add me on Discord `barthy.koeln#2946`. Sorry if I seem pushy, I'm just working on a Payload + Astro project right now :)

Not pushy at all! I appreciate the help :slightly_smiling_face:

**Comment by barthy-koeln** [comment]
> no open source headless CMS with good OpenAPI support 

Agreed, and your version outputs the cleanest so far. Not much bloat, readable names, yay.

> ideas or requests

I think the current code base needs some clean-up and restructuring to make further development easier.
There are two main pain points for me:

* Split `src/openapiSpec.ts` into smaller modules
  * string manipulation functions
  * schema name functions
  * schema generator functions
  * …
* Enforce a code style and best practices
  * indexer access vs. dot-notation
  * enforce multi-line objects
  * … 

I'd be happy to code a suggestion if you're available for reviews.

**Comment by Teyras** [comment]
> Agreed, and your version outputs the cleanest so far. Not much bloat, readable names, yay.

I'm excited to hear that!

> I'd be happy to code a suggestion if you're available for reviews.

Sure! 

> * Split `src/openapiSpec.ts` into smaller modules
>   
>   * string manipulation functions
>   * schema name functions
>   * schema generator functions
>   * …

Yeah, the transition from proof-of-concept to a maintainable package is not 100% yet. 

> * Enforce a code style and best practices
>   
>   * indexer access vs. dot-notation
>   * enforce multi-line objects
>   * …

I usually just run `eslint --fix` and `prettier --write` on the codebase and call it a day, but I understand that it's not the most welcoming way for community contributors. 

Also, there is now https://github.com/payloadcms/eslint-config - it may be worth it to use it.

**Comment by barthy-koeln** [comment]
Hi again, I'm unsure when or how I'll have the time to work more on this. I would submit another PR anyway.

Can we merge and release this one?

**Comment by Teyras** [comment]
> Hi again, I'm unsure when or how I'll have the time to work more on this. I would submit another PR anyway.
> 
> Can we merge and release this one?

Hi, done, merged, released. I may have some time to work on the plugin next weekend. If it actually happens, I'll take care of that enum as well.

**Pull request merged**